torchtitan/train.py

```diff
@@ -718,6 +723,13 @@ def main(trainer_class: type[Trainer]) -> None:
     try:
         trainer = trainer_class(config)
 
+        # TODO(local_tensor): Remove this special case once LocalTensor supports
+        # init_weights(). In local tensor mode, skip training/checkpointing as the
+        # model is not fully initialized
+        if config.comm.local_tensor_mode:
```

Comment:
probably a naive question. what's the advantage of localtensor over faketensor if we're just running the parallelization setup code?

localtensor would do the actual compute and give correct results, but it would run (more slowly) on a single gpu since it simulates each rank's operations.  fake tensor would skip all the compute and run (more quickly, i think) on a single CPU.

Do we intend to run with numerics or just smoke-test that we don't get API errors along the way to setup?